tests/test_main.py

```diff
@@ -170,11 +170,19 @@ def test_metadata_loads_egg_info(self):
         assert meta['Description'] == 'pôrˈtend'
 
 
-class DiscoveryTests(fixtures.EggInfoPkg, fixtures.DistInfoPkg, unittest.TestCase):
+class DiscoveryTests(
+    fixtures.EggInfoPkg,
+    fixtures.EggInfoPkgPipInstalledNoModules,
+    fixtures.EggInfoPkgSourcesFallback,
```

Comment:
I worry a bit about the complexity that this adds. I'm not sure what to do about it at this moment, but my initial instict is that these concerns are too entangled (suggesting that EggInfo and DistInfo were too entangled before, but since it was just two, it was acceptable). Maybe this approach is suitable, but I'll leave this flag for now to consider.

Reply:
Maybe it would look better to restructure the testdata (i.e. the various `DistInfo*` and `EggInfo*` classes in the `fixtures` module) using `@pytest.mark.parametrize(...)` instead of passing them as base classes to these tests? A major difference would then be that the tests are run against each of the package types in turn, instead of running them _once_ testing all packages types in one go. But there might be good reasons for why the tests are structured the current way?

Reply:
That's right. Although the test suite here leverages pytest, these same tests are ported into CPython where pytest is unavailable, so only unittest functionality is available. That's the motivation for these aberrant patterns. Elsewhere, I've started re-implementing `parameterization` similar to what pytest provides. I'm going to kick this can down the road for now and leave a refactoring for a later consideration.